x-pack/plugins/ml/public/components/annotations/annotation_flyout/index.tsx

```diff
@@ -113,6 +114,45 @@ class AnnotationFlyoutIntl extends Component<CommonProps & Props & InjectedIntlP
     this.setState({ isDeleteModalVisible: false });
   };
 
+  public validateAnnotationText = () => {
+    // Validates the entered text, returing an array of error messages
+    // for display in the form. An empty array is returned if the text is valid.
+    const { annotation, intl } = this.props;
+    const errors: string[] = [];
+    if (annotation === null) {
+      return errors;
+    }
+
+    if (annotation.annotation.trim().length === 0) {
```

Comment:
Is it possible for `annotation` here to be `undefined`? If so, might be useful to add a check for it being defined before accessing the `annotation` property.

Reply:
I don't think it should ever be `undefined`. It will either have a value, when editing an annotation, or it will be `null` when the flyout is closed.